src/services/GetRepositoriesService.ts

```diff
@@ -1,9 +1,16 @@
 import api from './api';
+import AppError from '../errors/AppError';
 import Repository from '../entities/Repository';
 
 class GetRepositoriesService {
   public async execute(username: string): Promise<Repository[]> {
-    const response = await api.get(`/users/${username}/repos`);
+    let response;
+    try {
+      response = await api.get(`/users/${username}/repos`);
+    } catch (error) {
+      const { data, status } = error.response;
```

Comment:
Esses tratamentos de erros devem ser mais genéricos. Você que deve definir. Por exemplo.
O provavel erro que deve dar aqui, é que o usuário não foi encontrado na API do github.
Então você lança uma exception, com o código padrão, e coloca uma mensagem do tipo "Github user is not valid".
Isso se repete nos outros tratamentos.

Reply:
@AntonioNtV  Eu pensei sobre isso, mas imaginei que não precisava já que o erro de usuário inválido vai ser obtido com o UserService que é o primeiro service a ser executado, então nesse local só podem ser outros erros. Porém, realmente esse service pode ser usado sem o UserService e a informação de erro vai ficar genérica.